stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/state/ChannelMutableState.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.offline.plugin.state
+
+import io.getstream.chat.android.client.events.ChatEvent
+import io.getstream.chat.android.client.models.ChannelUserRead
+import io.getstream.chat.android.client.models.Config
+import io.getstream.chat.android.client.models.Member
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.offline.model.channel.ChannelData
+import io.getstream.chat.android.offline.plugin.state.channel.ChannelState
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import java.util.Date
+
+@Suppress("VariableNaming")
+internal interface ChannelMutableState : ChannelState {
+
+    val _messages: MutableStateFlow<Map<String, Message>>
+    val _watcherCount: MutableStateFlow<Int>
+    val _typing: MutableStateFlow<Map<String, ChatEvent>>
+    val _reads: MutableStateFlow<Map<String, ChannelUserRead>>
+    val _read: MutableStateFlow<ChannelUserRead?>
+    val _endOfNewerMessages: MutableStateFlow<Boolean>
+    val _endOfOlderMessages: MutableStateFlow<Boolean>
+    val _loading: MutableStateFlow<Boolean>
+    val _hidden: MutableStateFlow<Boolean>
+    val _muted: MutableStateFlow<Boolean>
+    val _watchers: MutableStateFlow<Map<String, User>>
+    val _members: MutableStateFlow<Map<String, Member>>
+    val _loadingOlderMessages: MutableStateFlow<Boolean>
+    val _loadingNewerMessages: MutableStateFlow<Boolean>
+    val _channelData: MutableStateFlow<ChannelData?>
+    val _oldMessages: MutableStateFlow<Map<String, Message>>
+    val lastMessageAt: MutableStateFlow<Date?>
+    val _repliedMessage: MutableStateFlow<Message?>
+    val _unreadCount: MutableStateFlow<Int>
+    val _membersCount: MutableStateFlow<Int>
+    val _insideSearch: MutableStateFlow<Boolean>
+
+    /** Channel config data. */
+    val _channelConfig: MutableStateFlow<Config>
```

Comment:
Hey @leandroBorgesFerreira 👋

Could we avoid exposing `MutableStateFlow` and probably replace it with setters as we've done in `MutableGlobalState`?